src/internet_identity/src/assets.rs

```diff
@@ -86,61 +78,104 @@ pub fn init_assets() {
-            "/".to_string(),
-            index_html.clone(),
-            ContentEncoding::Identity,
-            ContentType::HTML,
-        ),
-        (
-            "/about".to_string(),
-            about_html,
-            ContentEncoding::Identity,
-            ContentType::HTML,
-        ),
-        (
-            "/index.html".to_string(),
-            index_html,
-            ContentEncoding::Identity,
-            ContentType::HTML,
-        ),
-        (
-            "/index.js".to_string(),
-            include_bytes!("../../../dist/index.js.gz").to_vec(),
-            ContentEncoding::GZip,
-            ContentType::JS,
-        ),
-        (
-            "/index.css".to_string(),
-            include_bytes!("../../../dist/index.css").to_vec(),
-            ContentEncoding::Identity,
-            ContentType::CSS,
-        ),
-        (
-            "/loader.webp".to_string(),
-            include_bytes!("../../../dist/loader.webp").to_vec(),
-            ContentEncoding::Identity,
-            ContentType::WEBP,
-        ),
-        (
-            "/favicon.ico".to_string(),
-            include_bytes!("../../../dist/favicon.ico").to_vec(),
-            ContentEncoding::Identity,
-            ContentType::ICO,
-        ),
-        // Required to make II available on the identity.internetcomputer.org domain.
-        // See https://github.com/r-birkner/portal/blob/rjb/custom-domains-docs-v2/docs/developer-docs/production/custom-domain/custom-domain.md#custom-domains-on-the-boundary-nodes
-        (
-            "/.well-known/ic-domains".to_string(),
-            b"identity.internetcomputer.org".to_vec(),
-            ContentEncoding::Identity,
-            ContentType::OCTETSTREAM,
-        ),
-    ]
+static ASSET_DIR: Dir<'_> = include_dir!("$CARGO_MANIFEST_DIR/../../dist");
+
+// Gets the static assets. All static assets are prepared only once (like injecting the canister ID).
+fn get_static_assets() -> Vec<(String, Vec<u8>, ContentEncoding, ContentType)> {
+    let mut assets = collect_assets_recursive(&ASSET_DIR);
+
+    // Required to make II available on the identity.internetcomputer.org domain.
+    // See https://internetcomputer.org/docs/current/developer-docs/production/custom-domain/#custom-domains-on-the-boundary-nodes
+    assets.push((
+        "/.well-known/ic-domains".to_string(),
+        b"identity.internetcomputer.org".to_vec(),
+        ContentEncoding::Identity,
+        ContentType::OCTETSTREAM,
+    ));
+
+    assets
+}
+
+fn collect_assets_recursive(dir: &Dir) -> Vec<(String, Vec<u8>, ContentEncoding, ContentType)> {
+    let mut assets = collect_assets_from_dir(dir);
+    for subdir in dir.dirs() {
+        assets.extend(collect_assets_recursive(subdir).into_iter());
+    }
+    assets
+}
+
+fn collect_assets_from_dir(dir: &Dir) -> Vec<(String, Vec<u8>, ContentEncoding, ContentType)> {
+    let mut assets: Vec<(String, Vec<u8>, ContentEncoding, ContentType)> = vec![];
+    for asset in dir.files() {
```

Comment:
do you know how this behaves with symlinks? I guess we'll want to symlink `about.html` to `index.html`

Reply:
Unfortunately this causes the `include_dir!` macro to panic. We would first need to add support in the library for that.

Reply:
ah, that's a shame. Maybe we can deduplicate the assets afterwards (and store them in a CAS) if necessary, but we'll worry about this only if the wasm module gets too big.